### DISCUSSION TITLE
json_type() function via PB api - query whether json field key exists

### DISCUSSION THREAD

**Discussion by ryanrussell** [pocketbase/pocketbase#5796] [Q&A] [answered]
## Is there a way to use the `JSON_TYPE()` function via the pb api?

Use case: `json_field` with optional numeric keys. Instead of zero, the key does not exist in the `json_field`. I want to query to check if a given key exists in `json_field`.

Consider 2 example `json_field` rows:
```json
  {"id": 1, "json_object_key": 5, ...},
  {"id": 2, ...} // json_object_key does not exist instead of 0 
```

### A working raw sqlite query is:
```sql
SELECT *
FROM table
WHERE JSON_TYPE(json_field, '$.json_object_key') IS NULL;
```

This works as expected with `*`, `count()`, `IS NULL`, `IS NOT NULL`, etc directly via sqlite.

### This doesn't seem to work via pb. 

When `json_field.json_object_key` exists, you can simply query like this via pb api:
```sql
json_field.json_object_key >= 1
```
#### but when `json_field.json_object_key` is null, this doesn't work:
```sql
JSON_TYPE(json_field, '$.json_object_key') IS NULL
```

Sidestepping the api to do some queries manually via sqlite works, but feels like a brittle solution.

Is detecting whether a key exists in a json field possible via the pb api?

**Comment by ganigeorgiev** [comment, marked as answer]
There is no support for calling directly the SQLite raw JSON functions using the Web API filters or rules.

I personally don't have a need for it and I don't understand the use case where this would be useful, so for the time being it remains out of the scope for v1.

In general you can write plain SQL queries when creating [View collections](https://pocketbase.io/docs/collections/#view-collection).

Depending on your use case, it is not exactly the same as checking whether the key exists, an alternative could be to compare the missing key value against `null` (e.g. `json_field.json_object_key = null`).

**Comment by ryanrussell** [comment]
Got it. Thanks
